### PR TITLE
further the Die abstraction work

### PR DESCRIPTION
**Pull request by grimreaper**
This furthers the die abstraction work but moving the addition of new values out of the Die and into a DieModifier.

In the old world there are multiple levels of abstraction:
Die, Dice, DieEx, SystemDie
all of which provide very specific features.

In the new world these will all be replaced by a DiceConfiguration which is just a sequence of filters

### CODE REVIEW

**Comment by cpmeister** [comment]
This branch has conflicts

**Comment by grimreaper** [comment]
fixed

File: code/src/java/gmgen/plugin/dice/AppendModifier.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+		int[] newResults = new int[count + in.length];
+		System.arraycopy(in, 0, newResults, 0, in.length);
+		for (int i = 0; i < count; ++i)
+		{
+			int thisRoll = rand.nextInt(max) + 1;
+			newResults[in.length + i] = thisRoll;
+		}
+		return newResults;
+	}
+}
```
Comment (cpmeister) [inline code/src/java/gmgen/plugin/dice/AppendModifier.java]: This file is missing the license header.

Comment (grimreaper) [inline code/src/java/gmgen/plugin/dice/AppendModifier.java]: adding license headers to all of the new files

Comment (cpmeister) [inline code/src/java/gmgen/plugin/dice/AppendModifier.java]: Thanks I forgot to notice the ones you added prior

**Pull request merged**